icechunk/src/session.rs

```diff
@@ -1375,52 +1456,81 @@ impl<'a> FlushProcess<'a> {
         asset_manager: Arc<AssetManager>,
         change_set: &'a ChangeSet,
         parent_id: &'a SnapshotId,
+        config: &'a RepositoryConfig,
     ) -> Self {
         Self {
             asset_manager,
             change_set,
             parent_id,
+            config,
             manifest_refs: Default::default(),
             manifest_files: Default::default(),
         }
     }
 
+    async fn write_manifest_from_iterator(
+        &mut self,
+        node_id: &NodeId,
+        chunks: impl Stream<Item = SessionResult<ChunkInfo>>,
+        splits: ManifestSplits,
+    ) -> SessionResult<()> {
+        // TODO: think about optimizing writes to manifests
+        // TODO: add benchmarks
+        let split_refs = chunks
```

Comment:
I wonder is there is a way to do this using less memory. For example, should we "classify" only the ChunkIndices instead of the whole ChunkInfo which is much larger? Then we use the ChangeSet or the previous manifest to get the contents of each index. Not sure if it works, but something to think about.

But bottom line, maybe keeping track of the shard each chunk belongs to in the manifest is not hard and that would help a lot with transient memory.